lib/active_model/serializer/caching.rb

```diff
@@ -227,28 +227,34 @@ def fetch(adapter_instance, cache_options = serializer_class._cache_options)
         end
       end
 
+      def non_cached_attributes(adapter_instance, fields)
+        non_cached_fields = fields[:non_cached].dup
+        non_cached_hash = attributes(non_cached_fields, true)
+        include_directive = JSONAPI::IncludeDirective.new(non_cached_fields - non_cached_hash.keys)
+        non_cached_hash.merge! resource_relationships({}, { include_directive: include_directive }, adapter_instance)
+        non_cached_hash
+      end
+
```

Comment:
this looks like it was removed from `fetch_attributes_fragment` unchanged? was it a rubocop thing?  The method needs refactor, I agree.. so sell me on why it's better with the `non_cached_hash` code moved to its own method (but not the `cached_hash`)?

Also, when adding methods, since this is in such flux, please prepend with comment `# @api private` and place them after where they are used